cognite/client/_api/simulators/simulator_routines.py

```diff
@@ -0,0 +1,134 @@
+
+    def list(
+        self, limit: int = DEFAULT_LIMIT_READ, filter: SimulatorRoutinesFilter | dict[str, Any] | None = None
+    ) -> SimulatorRoutineList:
+        """`Filter Simulators <https://api-docs.cognite.com/20230101-alpha/tag/Simulators/operation/filter_simulators_simulators_list_post>`_
+
+        List simulators
+
+        Args:
+            limit (int): The maximum number of simulators to return. Defaults to 100.
+            filter (SimulatorRoutinesFilter | dict[str, Any] | None): The filter to narrow down simulator routines.
+
+        Returns:
+            SimulatorRoutineList: List of simulator routines
+
+        Examples:
+
+            List simulators:
+
+                    >>> from cognite.client import CogniteClient
+                    >>> client = CogniteClient()
+                    >>> res = client.simulators.list_routines()
+
+        """
+        self._warning.warn()
+        return self._list(
+            method="POST",
+            limit=limit,
+            url_path="/simulators/routines/list",
+            resource_cls=SimulatorRoutine,
+            list_cls=SimulatorRoutineList,
+            headers={"cdf-version": "beta"},
+            filter=filter.dump()
+            if isinstance(filter, SimulatorRoutinesFilter)
+            else filter
+            if isinstance(filter, dict)
+            else None,  # fix this
+        )
+
+    def list_revisions(
+        self, limit: int = DEFAULT_LIMIT_READ, filter: SimulatorRoutineRevisionsFilter | dict[str, Any] | None = None
+    ) -> SimulatorRoutineRevisionsList:
+        """`Filter Simulators <https://api-docs.cognite.com/20230101-alpha/tag/Simulators/operation/filter_simulators_simulators_list_post>`_
+
+        List simulators
+
+        Args:
+            limit (int): The maximum number of simulators to return. Defaults to 25. Set to -1, float("inf") or None
+            filter (SimulatorRoutineRevisionsFilter | dict[str, Any] | None): The filter to narrow down simulator routine revisions.
+
+        Returns:
+            SimulatorRoutineRevisionsList: List of simulator routines
+
+        Examples:
+
+            List simulators:
+
+                    >>> from cognite.client import CogniteClient
+                    >>> client = CogniteClient()
+                    >>> res = client.simulators.list_revisions()
+
+        """
+        self._warning.warn()
+        return self._list(
+            method="POST",
+            limit=limit,
+            url_path="/simulators/routines/revisions/list",
+            resource_cls=SimulatorRoutineRevision,
+            list_cls=SimulatorRoutineRevisionsList,
+            headers={"cdf-version": "beta"},
+            filter=filter.dump()
+            if isinstance(filter, SimulatorRoutineRevisionsFilter)
+            else filter
+            if isinstance(filter, dict)
+            else None,  # fix this
+        )
+
+    def retrieve_revision(
```

Comment:
This one should be as well under `revisions.retrieve`, allowing for a call like `client.simulators.routines.revisions.retrieve()`.

Reply:
This also needs to be refactored to allow retrieving multiple revisions